pkg/collector/worker/worker.go

```diff
@@ -141,6 +147,13 @@ func (w *Worker) Run() {
 			continue
 		}
 
+		if !w.shouldRunIntegrationInstance(check) {
```

Comment:
Would it be possible to move this block before `checksTracker.AddCheck` block directly above? Then you wouldn't need to call `DeleteCheck` here, and we don't need to deduplicate runs of non-running check.

Reply:
Good idea, fixed here https://github.com/DataDog/datadog-agent/pull/31186/commits/5ed60bbfe99f9fa395e9f6cd87f830836a7d29a9